main.go

```diff
@@ -1,91 +1,72 @@
 package main
 
 import (
-	"fmt"
-	"net"
+	"log"
 	"os"
-	"strconv"
-	"strings"
-	"time"
 
-	"github.com/drone/drone-go/drone"
-	"github.com/drone/drone-go/plugin"
-	"golang.org/x/crypto/ssh"
+	_ "github.com/joho/godotenv/autoload"
+	"github.com/urfave/cli"
 )
 
-var (
-	buildCommit string
-)
+var version string // build number set at compile-time
 
 func main() {
-	fmt.Printf("Drone SSH Plugin built from %s\n", buildCommit)
-
-	workspace := drone.Workspace{}
-	vargs := Params{}
-
-	plugin.Param("workspace", &workspace)
-	plugin.Param("vargs", &vargs)
-	plugin.MustParse()
-
-	for i, host := range vargs.Host.Slice() {
-		err := run(workspace.Keys, &vargs, host)
-
-		if err != nil {
-			fmt.Println(err)
-			os.Exit(1)
-		}
-
-		if vargs.Sleep != 0 && i != vargs.Host.Len()-1 {
-			fmt.Printf("$ sleep %d\n", vargs.Sleep)
-			time.Sleep(time.Duration(vargs.Sleep) * time.Second)
-		}
-	}
-}
-
-func run(key *drone.Key, params *Params, host string) error {
-	if params.Login == "" {
-		params.Login = "root"
-	}
-
-	if params.Port == 0 {
-		params.Port = 22
-	}
-
-	addr := net.JoinHostPort(
-		host,
-		strconv.Itoa(params.Port),
-	)
-
-	fmt.Printf("$ ssh %s@%s -p %d\n", params.Login, addr, params.Port)
-	signer, err := ssh.ParsePrivateKey([]byte(key.Private))
-
-	if err != nil {
-		return fmt.Errorf("Error: Failed to parse private key. %s", err)
-	}
-
-	config := &ssh.ClientConfig{
-		User: params.Login,
-		Auth: []ssh.AuthMethod{
-			ssh.PublicKeys(signer),
+	app := cli.NewApp()
+	app.Name = "ssh plugin"
+	app.Usage = "ssh plugin"
+	app.Action = run
+	app.Version = version
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:   "sshkey",
+			Usage:  "private ssh key",
+			EnvVar: "PLUGIN_SSHKEY",
```

Comment:
let's use `PLUGIN_SSH_KEY` (`ssh_key` in yaml). We should also really do `PLUGIN_SSK_KEY,SSH_KEY` to support secrets which are not injected with the `PLUGIN_` prefix